### PR TITLE
Processing data

### PR DESCRIPTION
**Pull request by roshan-bajgain**
### Blog App - processing data in models
- Create at least one user by running the following code
- Create at least 4 posts written by one of the users you created by running the following code
- Create at least 6 posts comments for one of the posts you created by running the following code
- Create custom methods for:
  - Users
  - Posts
  - Comments
  - Likes
- you will need to make current user data available to all controllers.

### CODE REVIEW

File: app/models/like.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class Like < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+
+  def update_comments_counter
+    post.increment!(:like_counter)
+  end
```
Comment (Mihndim2020) [inline app/models/like.rb]: - [OPTIONAL] I think this should be `update_likes_counter`, please kindly fix the typing error.

File: app/models/comment.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class Comment < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+
```
Comment (Mihndim2020) [inline app/models/comment.rb:4]: - [OPTIONAL] Please you can  can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_comments_counter` method every time a new comment has been saved to database, then make the `update_comments_counter` method private.

File: app/models/like.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+class Like < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+
```
Comment (Mihndim2020) [inline app/models/like.rb:4]: - [OPTIONAL] Please you can  can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_likes_counter` method every time a new like has been saved to database, then make the `update_likes_counter` method private.

File: app/models/post.rb [hunk 4]
```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments
+  has_many :likes
+
```
Comment (Mihndim2020) [inline app/models/post.rb:5]: - [OPTIONAL] Please you can  can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_posts_counter` method every time a new post has been saved to database, then make the `update_posts_counter` method private. Something like so: 
```ruby
  after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end
```

**Review by Mihndim2020 — APPROVED** [review]
# STATUS: APPROVED!!!  🎉 🎉 🎉 :champagne: :champagne: :champagne:
### Hi @roshan-bajgain, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 :champagne: :champagne: :champagne:

Congratulations-Memes-FEATURED

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by roshan-bajgain** [comment]
@Mihndim2020 Thank for you suggestion I have implemented it

**Pull request merged**